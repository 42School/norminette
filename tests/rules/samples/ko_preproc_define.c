#define toto f(x)
#define A AB+AB
#define PRINTF printf()
#define PRINTF printf
#define TOTO TATA 12
 #define TOTo TATa