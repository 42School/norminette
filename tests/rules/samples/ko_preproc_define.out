[36mko_preproc_define.c[0m - [32mIsPreprocessorStatement[0m In "GlobalScope" from "None" line 1":
		<HASH> <IDENTIFIER=define> <SPACE> <IDENTIFIER=toto> <SPACE> <IDENTIFIER=f> <LPARENTHESIS> <IDENTIFIER=x> <RPARENTHESIS> <NEWLINE>
[36mko_preproc_define.c[0m - [32mIsPreprocessorStatement[0m In "GlobalScope" from "None" line 2":
		<HASH> <IDENTIFIER=define> <SPACE> <IDENTIFIER=A> <SPACE> <IDENTIFIER=AB> <PLUS> <IDENTIFIER=AB> <NEWLINE>
[36mko_preproc_define.c[0m - [32mIsPreprocessorStatement[0m In "GlobalScope" from "None" line 3":
		<HASH> <IDENTIFIER=define> <SPACE> <IDENTIFIER=PRINTF> <SPACE> <IDENTIFIER=printf> <LPARENTHESIS> <RPARENTHESIS> <NEWLINE>
[36mko_preproc_define.c[0m - [32mIsPreprocessorStatement[0m In "GlobalScope" from "None" line 4":
		<HASH> <IDENTIFIER=define> <SPACE> <IDENTIFIER=PRINTF> <SPACE> <IDENTIFIER=printf> <NEWLINE>
[36mko_preproc_define.c[0m - [32mIsPreprocessorStatement[0m In "GlobalScope" from "None" line 5":
		<HASH> <IDENTIFIER=define> <SPACE> <IDENTIFIER=TOTO> <SPACE> <IDENTIFIER=TATA> <SPACE> <CONSTANT=12> <NEWLINE>
[36mko_preproc_define.c[0m - [32mIsPreprocessorStatement[0m In "GlobalScope" from "None" line 6":
		<SPACE> <HASH> <IDENTIFIER=define> <SPACE> <IDENTIFIER=TOTo> <SPACE> <IDENTIFIER=TATa> 
ko_preproc_define.c: Error!
Error: INVALID_HEADER       (line:   1, col:   1):	Missing or invalid 42 header
Error: MACRO_NAME_CAPITAL   (line:   1, col:   9):	Macro name must be capitalized
Error: PREPROC_CONSTANT     (line:   1, col:  15):	Preprocessor statement must only contain constant defines
Error: PREPROC_CONSTANT     (line:   2, col:  13):	Preprocessor statement must only contain constant defines
Error: PREPROC_CONSTANT     (line:   3, col:  22):	Preprocessor statement must only contain constant defines
Error: PREPROC_CONSTANT     (line:   5, col:  19):	Preprocessor statement must only contain constant defines
Error: SPACE_REPLACE_TAB    (line:   6, col:   2):	Found space when expecting tab
Error: MACRO_NAME_CAPITAL   (line:   6, col:  10):	Macro name must be capitalized
